mmaction/core/evaluation/eval_hooks.py

```diff
@@ -25,20 +23,26 @@ class EpochEvalHook(Hook):
             evaluation before the training starts if ``start`` <= the resuming
             epoch. If None, whether to evaluate is merely decided by
             ``interval``. Default: None.
-        interval (int): Evaluation interval (by epochs). Default: 1.
-        save_best (bool): Whether to save best checkpoint during evaluation.
-            Default: True.
-        key_indicator (str | None): Key indicator to measure the best
-            checkpoint during evaluation when ``save_best`` is set to True.
+        interval (int): Evaluation interval. Default: 1.
+        by_epoch (bool): Determine perform evaluation by epoch or by iteration.
+            If set to True, it will perform by epoch. Otherwise, by iteration.
+            default: True.
+        save_best (str | None, optional): If a metric is specified, it would
+            measure the best checkpoint during evaluation. The information
+            about best checkpoint would be save in best.json.
             Options are the evaluation metrics to the test dataset. e.g.,
              ``top1_acc``, ``top5_acc``, ``mean_class_accuracy``,
             ``mean_average_precision``, ``mmit_mean_average_precision``
             for action recognition dataset (RawframeDataset and VideoDataset).
             ``AR@AN``, ``auc`` for action localization dataset
-            (ActivityNetDataset). Default: `top1_acc`.
-        rule (str | None): Comparison rule for best score. Options are None,
-            'greater' and 'less'. If set to None, it will infer a reasonable
-            rule. Default: 'None'.
+            (ActivityNetDataset). If ``save_best`` is ``auto``, the first key
+             of the returned ``OrderedDict`` result will be used. The interval
+             of ``CheckpointHook`` should device EvalHook. Default: None.
```

Comment:
what does `device EvalHook` mean? is this a typo?

Actually, I did some digging about hooks in mmcv. The best ckpt saved in `EvalHook` is a symlink of `runner.meta['hook_msgs']['last_ckpt']`, which is generated by `CheckpointHook`. So if checkpoint hook interval is 5 & eval hook interval is 1 & best ckpt is from epoch no.7, then the best ckpt we saved is from epoch no. 5.

Is this the case? Can we just save the ckpt in EvalHook?

Reply:
Save the checkpoint if the symlink can not be created ? @innerlee @irvingzhang0512

Reply:
I prefer saving ckpt instead of creating symlink. Since we don't have early stopping by default, if we set `max_keep_ckpts` in checkpoint hook, the best ckpt we save may be removed.

By the way, it seems that current codes doesn't remove previous best ckpt. Say we train for 10 epochs, and val accuracy increases epoch by epoch. By using our eval hook, 10 best ckpt symlink left..

Reply:
use cp instead of symlink now. Thanks!